### PR TITLE
Docs: Updated README to include crates information

### PR DESCRIPTION
**Pull request by naveensrinivasan**
Included crates information into README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,6 +33,15 @@ production-worthy implementation in 2021.
 Communications for Rust-Lightning and Lightning Development Kit happens through
 [LDK slack](http://lightningdevkit.org/).
 
+Crates
+-----------
+1. [lightning](./lightning)
+1. [lightning-background-processor](./lightning-background-processor)
```
Comment (TheBlueMatt) [inline README.md:39]: Maybe change the numbers :). But, should we say a brief word about each? Should be able to copy and tweak the information in each directory's `Cargo.toml`.

Comment (naveensrinivasan) [reply README.md:39]: I can change the numbers. I was also thinking about adding some description but didn't know where to look for it.

Comment (TheBlueMatt) [reply README.md:39]: Look at the `Cargo.toml` and the crate's documentation (ie the big `//!` comment at the top of `lib.rs`).

Comment (naveensrinivasan) [reply README.md]: I used the `Cargo.toml` docs. Let me know.

File: README.md [hunk 2]
```diff
@@ -33,6 +33,25 @@ production-worthy implementation in 2021.
 Communications for Rust-Lightning and Lightning Development Kit happens through
 [LDK slack](http://lightningdevkit.org/).
 
+Crates
+-----------
+1. [lightning](./lightning)   
+A Bitcoin Lightning library in Rust.
```
Comment (TheBlueMatt) [inline README.md:39]: This feels strange - the whole repo is a "Bitcoin lightning library" - maybe for lightning reword it to say something about "The Core of the LDK library, implements the lightning protocol, channel state machine, and on-chain logic. Supports no-std and exposes on relatively low-level interfaces".

File: README.md [hunk 3]
```diff
@@ -33,6 +33,25 @@ production-worthy implementation in 2021.
...
+Does most of the hard work, without implying a specific runtime, requiring clients implement basic 
+network logic, chain interactions and disk storage.
+2. [lightning-background-processor](./lightning-background-processor)
+Utilities to perform required background tasks for Rust Lightning.
```
Comment (TheBlueMatt) [inline README.md:43]: Does the md still render correctly if we indent the lines a bit? Its nice when .md files read well even when not rendered, IMO.

File: README.md [hunk 4]
```diff
@@ -33,6 +33,25 @@ production-worthy implementation in 2021.
 Communications for Rust-Lightning and Lightning Development Kit happens through
 [LDK slack](http://lightningdevkit.org/).
 
+Crates
+-----------
+1. [lightning](./lightning)
```
Comment (TheBlueMatt) [inline README.md:38]: nit: several lines have trailing whitespace, including this one.

Comment (naveensrinivasan) [reply README.md:38]: It was only for one line AFAIK. I fixed it. Let me know if there are other\.

File: README.md (hunk 2 above)
Comment (naveensrinivasan) [reply README.md]: Thanks, updated it.

File: README.md (hunk 3 above)
Comment (naveensrinivasan) [reply README.md]: Fixed the indetation.

**Review by TheBlueMatt — COMMENTED** [review]
LGTM, can you squash the commits here?

**Comment by naveensrinivasan** [comment]
> LGTM, can you squash the commits here?

Done. Thanks!

**Review by TheBlueMatt — APPROVED** [review]

**Review by valentinewallace — APPROVED** [review]
